spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/serviceregistry/ZookeeperAutoServiceRegistrationAutoConfiguration.java

```diff
@@ -42,8 +42,8 @@
 @ConditionalOnMissingBean(type = "org.springframework.cloud.zookeeper.discovery.ZookeeperLifecycle")
 @ConditionalOnZookeeperDiscoveryEnabled
 @ConditionalOnProperty(value = "spring.cloud.service-registry.auto-registration.enabled", matchIfMissing = true)
-@AutoConfigureAfter(ZookeeperServiceRegistryAutoConfiguration.class)
-@AutoConfigureBefore( {AutoServiceRegistrationAutoConfiguration.class, ZookeeperDiscoveryAutoConfiguration.class} )
+@AutoConfigureAfter( {AutoServiceRegistrationAutoConfiguration.class,ZookeeperServiceRegistryAutoConfiguration.class} )
```

Comment:
Why this change?

Reply:
Because now the `AutoServiceRegistrationProperties` bean will only be created once `AutoServiceRegistrationAutoConfiguration` is run.  See https://github.com/spring-cloud/spring-cloud-commons/pull/245/files#diff-ead0de8e4600f43bcaadeee88d39b828R16

Reply:
I don't see how that justifies mocinv from Before to After. `AutoServiceRegistrationAutoConfiguration` will fail with an exception because of this change.

Reply:
auto configuration boggles my mind.  The exception will only be thrown is `failfast` is set to true, which may explain why I dont see any exceptions being thrown.  However even with telling it to auto configure after `AutoServiceRegistrationAutoConfiguration`, it is still running before!  What?!

Reply:
@spencergibb do u know why `@ConditionalOnBean(AutoServiceRegistrationProperties.class)` is even on this class?  `AutoServiceRegistrationProperties` isn't used anywhere

Reply:
@spencergibb ping ^^^

Reply:
It's the marker bean that auto-registration was enabled

Reply:
That seems a little redundant since there is already a `@ConditionalOnProperty` for `spring.cloud.service-registry.auto-registration.enabled`.

Reply:
@spencergibb do you think we can remove the `@ConditionalOnBean(AutoServiceRegistrationProperties.class)` and keep the `@ConditionalOnProperty(spring.cloud.service-registry.auto-registration.enabled)`?